libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServiceCheckbox.tsx

```diff
@@ -0,0 +1,74 @@
+import { Checkbox, TextRegular } from '@monorepo/expo/shared/ui-components';
+import { StyleSheet, View } from 'react-native';
+import { ServiceEnum } from '../../apollo';
+
+interface IServiceCheckboxProps {
+  service: {
+    title: string;
+    enum: ServiceEnum;
+  };
+  noteId: string | undefined;
+  idx: number;
+  services: {
+    id: string | undefined;
+    enum: ServiceEnum | null;
+    markedForDeletion?: boolean;
+  }[];
+  setServices: (
+    services: {
+      enum: ServiceEnum | null;
+      id: string | undefined;
+      markedForDeletion?: boolean;
+    }[]
+  ) => void;
+}
+
+export default function ServiceCheckbox(props: IServiceCheckboxProps) {
+  const { service, idx, services, setServices } = props;
+
+  const serviceEntry = services.find((s) => s.enum === service.enum);
+  const isChecked = (serviceEntry && !serviceEntry.markedForDeletion) || false;
+
+  const handleCheck = () => {
```

Comment:
I know we use the pattern of passing the `setServices` (or setWhatever) function down to children, but seems like a lot of responsibility to give to a checkbox. Any reason it can't return a boolean and the parent take care of the updating the actual services list?

Reply:
as service checkbox is not reusable for other cases I didn't want to make parent bigger.

Reply:
ok. 
won't block on this, but I think that if parent component is too big then probably better to address that directly as opposed to moving responsibilities down to children.